cdc/model/changefeed.go

```diff
@@ -305,6 +310,57 @@ func (info *ChangeFeedInfo) fixState() {
 	}
 }
 
+// fixSinkProtocol attempts to fix protocol incompatible.
+// We no longer support the acceptance of protocols that are not known.
+// The ones that were already accepted need to be fixed.
+func (info *ChangeFeedInfo) fixSinkProtocol() {
+	sinkURIParsed, err := url.Parse(info.SinkURI)
+	if err != nil {
+		log.Warn("parse sink URI failed", zap.Error(err))
+		// SAFETY: It is safe to ignore this unresolvable sink URI here,
+		// as it is almost impossible for this to happen.
+		// If we ignore it when fixing it after it happens,
+		// it will expose the problem when starting the changefeed,
+		// which is easier to troubleshoot than reporting the error directly in the bootstrap process.
+		return
+	}
+	rawQuery := sinkURIParsed.Query()
+	protocolStr := rawQuery.Get(config.ProtocolKey)
+
+	needsFix := func(protocolStr string) bool {
```

Comment:
how about move this out to be a function?

Reply:
It seems only possible to use it here. So maybe it would be more appropriate to put it inside? It is the core logic of this function.